### PR TITLE
Feature/g0 4277 stage

### PR DESCRIPTION
**Pull request by prateek-giddh**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts [hunk 1]
```diff
@@ -1022,11 +1012,19 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
...
-    public calculateConversionRate(baseModel) {
+    /**
+     * Calculates conversion rate
+     *
+     * @param {*} baseModel Value to be converted
+     * @param {number} [customDecimalPaces] Optional custom decimal places (required for Rate as 4 digits are required for rate)
+     * @returns Converted rate
+     * @memberof NewLedgerEntryPanelComponent
+     */
+    public calculateConversionRate(baseModel: any, customDecimalPaces?: number): number {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts:1023]: Correct spelling of Places in customDecimalPlaces

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedger.vm.ts [hunk 2]
```diff
@@ -660,14 +663,22 @@ export class UpdateLedgerVm {
...
-    public calculateConversionRate(baseModel) {
+    /**
+     * Calculates conversion rate
+     *
+     * @param {*} baseModel Value to be converted
+     * @param {number} [customDecimalPaces] Optional custom decimal places (required for Rate as 4 digits are required for rate)
+     * @returns Converted rate
+     * @memberof UpdateLedgerVm
+     */
+    public calculateConversionRate(baseModel: any, customDecimalPaces?: number): number {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedger.vm.ts:674]: Correct spelling of Places in customDecimalPlaces

File: apps/web-giddh/src/app/shared/helpers/pipes/currencyPipe/currencyType.pipe.ts [hunk 3]
```diff
@@ -22,9 +23,9 @@ export class GiddhCurrencyPipe implements OnInit, OnDestroy, PipeTransform {
             this.store.select(p => p.settings.profile).pipe(takeUntil(this.destroyed$), distinctUntilKeyChanged('balanceDisplayFormat')).subscribe((o) => {
                 if (o && o.name) {
                     this._currencyNumberType = o.balanceDisplayFormat ? o.balanceDisplayFormat : 'IND_COMMA_SEPARATED';
-                    this._currencyDesimalType = o.balanceDecimalPlaces ? o.balanceDecimalPlaces : 0;
-                    if (this._currencyDesimalType) {
-                        localStorage.setItem('currencyDesimalType', this._currencyDesimalType.toString());
+                    this.currencyDecimalType = o.balanceDecimalPlaces ? o.balanceDecimalPlaces : 0;
+                    if (this.currencyDecimalType) {
+                        localStorage.setItem('currencyDesimalType', this.currencyDecimalType.toString());
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/helpers/pipes/currencyPipe/currencyType.pipe.ts:28]: Correct spelling of decimal

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedger.vm.ts (hunk 2 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedger.vm.ts]: Done

File: apps/web-giddh/src/app/shared/helpers/pipes/currencyPipe/currencyType.pipe.ts (hunk 3 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/shared/helpers/pipes/currencyPipe/currencyType.pipe.ts:28]: Done

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts (hunk 1 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts]: Done

**Pull request merged**